### PR TITLE
SEQNG-312: Display file id of already executed steps

### PR DESCRIPTION
**Pull request by cquiroz**
On the new seqexec we'd like to display the file id of already executed steps. This information is on the program and can be transferred to the seqexec when reading the configuration sequencep

### CODE REVIEW

File: bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala [hunk 1]
```diff
@@ -83,25 +88,43 @@ object SeqExecService {
       }
     }
 
+  private def extractName(obs: ISPObservation): TrySeq[String] = trySeq {
+    obs.getDataObject match {
+      case o: SPObservation => o.getTitle.right
```
Comment (swalker2m) [inline bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala]: `getTitle` is defined in the `ISPDataObject` from which `SPObservation` inherits, so you wouldn't need this match unless you just want to verify that the data object is in fact an `SPObservation`.  If it isn't though, nothing else will work anyway.

Comment (cquiroz) [reply bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala]: Right, do you think I should just call it on the `DataObject`?

File: bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala [hunk 2]
```diff
@@ -83,25 +88,43 @@ object SeqExecService {
...
   private def extractSequence(obs: ISPObservation): TrySeq[ConfigSequence] =
     catchingAll {
       ConfigBridge.extractSequence(obs, null, IDENTITY_MAP, true)
     }
 
+  private def extractExecutedDatsets(obs: ISPObservation): TrySeq[Map[Int, ExecutedDataset]] =
+    catchingAll {
+      Option(ObsLog.getIfExists(obs)) match {
+        case Some(obsLog) => obsLog.getAllDatasetRecords.asScala.map(_.exec.dataset).map { d =>
+          d.getIndex -> ExecutedDataset(Instant.ofEpochMilli(d.getTimestamp), d.getDhsFilename)
```
Comment (cquiroz) [inline bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala]: It seems that `getIndex` is the step where the file was executed right. There is another `index` on `DatasetLabel`. @swalker2m  Do you know if they are the same?

File: bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala (hunk 1 above)
Comment (swalker2m) [reply bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala]: Yeah, that's probably what I'd do.

File: bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala (hunk 2 above)
Comment (swalker2m) [reply bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala]: I usually key datasets by their full dataset label, which includes the index.  In reality in the current OCS these must be the same.  That's the source of innumerable problems that motivated the need for a new sequence model.

**Pull request merged**